File: java/src/main/java/winium/elements/desktop/Menu.java
package winium.elements.desktop;


import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.RemoteWebElement;

import java.util.HashMap;

public class Menu extends DesktopElement {
    public static final String FIND_MENU_ITEM = "findMenuItem";
    public static final String SELECT_MENU_ITEM = "selectMenuItem";

    private RemoteWebElement callMenuItemCommand(String command, String path) {
        HashMap<String, Object> parameters = new HashMap<String, Object>();
        parameters.put("id", this.getId());
        parameters.put("path", path);
        return this.createRemoteWebElementFromResponse(this.execute(command, parameters));

    }

    public Menu(WebElement element) {
        super(element);
    }

    public RemoteWebElement findItem(String path) {
        return this.callMenuItemCommand(FIND_MENU_ITEM, path);
    }

    public RemoteWebElement selectItem(String path) {
        return this.callMenuItemCommand(SELECT_MENU_ITEM, path);
    }
}

File: dotnet/src/Winium.Elements.Desktop/Menu.cs
namespace Winium.Elements.Desktop
{
    #region using

    using System.Collections.Generic;

    using OpenQA.Selenium;
    using OpenQA.Selenium.Remote;

    #endregion

    public class Menu : DesktopElement
    {
        #region Constants

        private const string FindMenuItem = "findMenuItem";

        private const string SelectMenuItem = "selectMenuItem";

        #endregion

        #region Constructors and Destructors

        public Menu(IWebElement element)
            : base(element)
        {
        }

        #endregion

        #region Public Methods and Operators

        public RemoteWebElement FindItem(string path)
        {
            return this.CallMenuItemCommand(FindMenuItem, path);
        }

        public RemoteWebElement SelectItem(string path)
        {
            return this.CallMenuItemCommand(SelectMenuItem, path);
        }

        #endregion

        #region Methods

        private RemoteWebElement CallMenuItemCommand(string command, string path)
        {
            var parameters = new Dictionary<string, object> { { "id", this.Id }, { "path", path } };
            var response = this.Execute(command, parameters);

            return this.CreateRemoteWebElementFromResponse(response);
        }

        #endregion
    }
}
